### PR TITLE
update service registry API and add nse name in requests

### PR DESCRIPTION
**Pull request by vladmoisescu**
Update serviceregistry API
Add NSE name and cluster name in ipam request
Add NSE name and cluster name in serviceregistry request

### CODE REVIEW

File: pkg/universal-cnf/config/config.go [hunk 1]
```diff
@@ -31,6 +32,12 @@ import (
 
 const (
 	PEER_NAME = "ucnf/peerName"
+	PodName = "nsepod.name"
+
+	nsePodNameVar = "NSE_POD_NAME"
+	nsePodName = "example"
+	clusterVar = "CLUSTER_NAME"
```
Comment (ana-moga) [inline pkg/universal-cnf/config/config.go]: maybe clusterEnv/nsePodNameEnv?

**Review by ana-moga — APPROVED** [review]

File: api/serviceregistry/serviceregistry.proto [hunk 2]
```diff
@@ -45,6 +45,7 @@ message WorkloadIdentifier {
   string cluster = 1;
   string podName = 2;
   string name = 3;
+  string nseName = 4;
```
Comment (mihmatache) [inline api/serviceregistry/serviceregistry.proto:48]: Why do we need both the `name` and 'nseName`?

Comment (tiswanso) [reply api/serviceregistry/serviceregistry.proto:48]: The idea is to track which NSE that a workload is connecting through so we can use the data to construct topology views.  The name is the "service" name that the workload is registering with and we use it for DNS reg.

File: api/serviceregistry/serviceregistry.proto [hunk 3]
```diff
@@ -45,6 +45,7 @@ message WorkloadIdentifier {
   string cluster = 1;
   string podName = 2;
   string name = 3;
```
Comment (adodon2go) [inline api/serviceregistry/serviceregistry.proto:47]: If 'name' is a serviceName here maybe we need to update it here and be more specific to not confuse its purpose.

File: pkg/universal-cnf/config/config.go (hunk 1 above)
Comment (adodon2go) [reply pkg/universal-cnf/config/config.go]: ClusterNameEnv

File: api/serviceregistry/serviceregistry.proto (hunk 3 above)
Comment (vladmoisescu) [reply api/serviceregistry/serviceregistry.proto:47]: It is not actually the service name. A service can be distributed between multiple clusters, like our helloworld test service.

File: pkg/universal-cnf/config/config.go (hunk 1 above)
Comment (vladmoisescu) [reply pkg/universal-cnf/config/config.go]: Done

**Review by adodon2go — APPROVED** [review]

**Review by tiswanso — APPROVED** [review]
lgtm

**Pull request merged**